server/v2/commands.go

```diff
@@ -11,26 +11,43 @@ import (
 
 	"github.com/spf13/cobra"
 
+	"cosmossdk.io/core/transaction"
 	"cosmossdk.io/log"
+	"github.com/spf13/viper"
 )
 
-func Commands(logger log.Logger, homePath string, modules ...ServerModule) (CLIConfig, error) {
+type NewServerModulesFunc func(*viper.Viper, log.Logger, transaction.Codec[transaction.Tx]) []ServerModule
+
+func Commands(rootCmd *cobra.Command, appCreator NewServerModulesFunc, codec transaction.Codec[transaction.Tx], logger log.Logger, homePath string, modules ...ServerModule) (CLIConfig, error) {
```

Comment:
appCreator should simply be simapp, like it is in simapp: e.g https://github.com/cosmos/cosmos-sdk/blob/5a8e9d7aa8ffd861f8339e12fcae8634c62ca405/simapp/simd/cmd/commands.go#L120